### ISSUE TITLE
Don't clear switchport description on ESI-controlled port

### ISSUE THREAD

**Issue by hakasapl** [CCI-MOC/esi#365]
Currently, when ESI provisions a port on a switch (Dell switch in this case), it looks like the port gets defaults - which means the description is also cleared. It would help if the description was preserved, or re-applied after the port is provisioned since it's useful to keep info about the node it's connected to etc. on the switch end.

I'm not all that familiar with the ESI code base, if this is possible what could this change look like?

**Comment by tzumainn** [comment]
Hi! This may be a bit difficult; it's the ansible playbooks in network-runner that actually manipulate the switch, and I had the impression that it was difficult to shut down a switchport and clean the existing configuration without simply resetting everything.

@naved001 as the resident expert on network-runner, do you have thoughts on this... ?

**Comment by naved001** [comment]
tl;dr: it may be possible.

In the case of the brocade switches, I issue the commands `no switchport` and `shutdown`, which removes most switchport configuration without having to know exactly what VLANs it is on. For example:

```
rbr-01(conf-if-te-3/0/23)# do show running-config interface TenGigabitEthernet 3/0/23
interface TenGigabitEthernet 3/0/23
 mtu 9216
 description neu-3-12-nic2
 switchport
 switchport mode trunk
 switchport trunk allowed vlan add 500-501
 no switchport trunk tag native-vlan
 switchport trunk native-vlan 500
 spanning-tree shutdown
 fabric isl enable
 fabric trunk enable
 shutdown
!
rbr-01(conf-if-te-3/0/23)# no switchport
rbr-01(conf-if-te-3/0/23)# shutdown
rbr-01(conf-if-te-3/0/23)# do show running-config interface TenGigabitEthernet 3/0/23
interface TenGigabitEthernet 3/0/23
 mtu 9216
 description neu-3-12-nic2
 fabric isl enable
 fabric trunk enable
 shutdown
!
```

And this preserves the other settings, like mtu, description etc. In fact, there's no command to put the interface in a default configuration. 

In case of the dell switch,

```
OCT4-SW-TORS(conf)#interface tengigabitethernet 1/40
OCT4-SW-TORS(conf-if-te-1/40)#no switchport
% Error: Te 1/40 Port is part of a non-default VLAN.
OCT4-SW-TORS(conf-if-te-1/40)#shutdown
```

I can't just disable switching and shutdown the port, I either have to go to each vlan and run `no untagged interface te 1/40` and then come back to it and run `no switchport` and `shutdown` which would then preserve any other configuration for this switchport.

The solution would to be read the switchport description before setting it to default config, and then set the description back again.  I guess in [the playbooks](https://github.com/ansible-network/network-runner/tree/devel/etc/ansible/roles/network-runner/providers/dellemc.os9.os9) that are calling the default method, if we can figure out how to read the switch description and store it in some variable within ansible and then set it back, then we may be able to do it.

**Comment by naved001** [comment]
we could use the [os9_command module](https://docs.ansible.com/ansible/latest/collections/dellemc/os9/os9_command_module.html#examples) to read the description.

```
OCT4-SW-TORS#show running-config interface tengigabitethernet 1/40 | grep description
 description something
```

**Comment by hakasapl** [comment]
Yes, that is something particularly annoying about Dell os9. I had to write this condition in my ansible-switch management playbook: https://github.com/CCI-MOC/ansible-switches/blob/main/filter_plugins/dell_os9.py#L489 for every time an interface switches L2 mode.

I had to essentially figure out what VLANs an interface was assigned and then remove them prior to changing the switchport status, it's a huge pain (in ansible-switch case I had to do that so that the interface doesn't go down when applying ansible config)

**Comment by joachimweyl** [comment]
done
